citest/json_predicate/binary_predicate.py

```diff
@@ -146,6 +150,83 @@ def __call__(self, context, value):
                            path_value=PathValue('', value), valid=valid)
 
 
+class DictMatchesPredicate(BinaryPredicate):
+  """Implements binary predicate comparison predicates against dict values.
+
+  The dictionary value is in the form {field : pred} where each element
+  in the dictionary is a predicate for validating that particular field.
+  A strict predicate means that exactly all specified fields must be present.
+  Otherwise this permits additional fields with arbitrary values.
+  """
+
+  @property
+  def strict(self):
+    """Whether only the specified fields may be present (True) or not."""
+    return self.__strict
+
+  def __init__(self, operand, **kwargs):
+    """Constructor."""
+    if not isinstance(operand, dict):
+      raise TypeError(
+          '{0} is not a dict: {1!r}'.format(operand.__class__, operand))
+    self.__strict = kwargs.pop('strict', False)
+    super(DictMatchesPredicate, self).__init__('Matches', operand, **kwargs)
+
+  def __call__(self, context, value):
+    """Implements Predicate interface.
+
+    context: [ExecutionContext] The execution context.
+    value: [dict] The dictionary to match against our operand.
```

Comment:
what's wrong with value. Passing a value into the __call__ method suggests it being an actual value. This method is part of the interface.
https://github.com/google/citest/blob/master/citest/json_predicate/predicate.py#L33